source/fluid_pressure_boundary_conditions/density.cc

```diff
@@ -0,0 +1,118 @@
+#include <aspect/melt.h>
+#include <utility>
+#include <limits>
+
+
+namespace aspect
+{
+  namespace FluidPressureBoundaryConditions
+  {
+
+    template <int dim>
+    void
+    Density<dim>::
+    fluid_pressure_gradient (
+      const MaterialModel::MaterialModelInputs<dim> &material_model_inputs,
+      const MaterialModel::MaterialModelOutputs<dim> &material_model_outputs,
+      std::vector<Tensor<1,dim> > &output
+    ) const
+    {
+      const MaterialModel::MeltOutputs<dim> *melt_outputs = material_model_outputs.template get_additional_output<MaterialModel::MeltOutputs<dim> >();
+      Assert(melt_outputs!=NULL, ExcMessage("Error, MeltOutputs are missing in fluid_pressure_gradient()"));
+      for (unsigned int q=0; q<output.size(); ++q)
+        {
+          const Tensor<1,dim> gravity = this->get_gravity_model().gravity_vector(material_model_inputs.position[q]);
+          output[q] = (((include_rho_s) ? material_model_outputs.densities[q] : 0.0)
+                       + ((include_rho_f) ? melt_outputs->fluid_densities[q] : 0.0))
+                      * gravity;
+        }
+    }
+
+    template <int dim>
+    void
+    Density<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Fluid Pressure Boundary Condition");
+      {
+        prm.enter_subsection("Density");
+        {
+          prm.declare_entry ("Include Fluid Density", "false", Patterns::Bool(),
+                             "Prescribing the gradient of the fluid pressure as "
+                             "fluid density times gravity causes melt to flow in "
+                             "with the same velocity as inflowing solid material, "
+                             "or no melt flowing in or out if the solid velocity "
+                             "normal to the boundary is zero.");
+          prm.declare_entry ("Include Solid Density", "true", Patterns::Bool(),
+                             "Prescribing the gradient of the fluid pressure as "
+                             "solid density times gravity (which is the lithostatic "
+                             "pressure) leads to approximately the same pressure in "
+                             "the melt as in the solid, so that fluid is only flowing "
+                             "in or out due to differences in dynamic pressure. ");
+        }
+        prm.leave_subsection ();
+      }
+      prm.leave_subsection ();
+    }
+
+
+    template <int dim>
+    void
+    Density<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Fluid Pressure Boundary Condition");
+      {
+        prm.enter_subsection("Density");
+        {
+          include_rho_f = prm.get_bool("Include Fluid Density");
+          include_rho_s = prm.get_bool("Include Solid Density");
+        }
+        prm.leave_subsection ();
+      }
+      prm.leave_subsection ();
+    }
+
+
+
+    template <int dim>
+    void
+    Density<dim>::initialize()
```

Comment:
You do not need to put this function here if it is empty. Just remove it (also from the header file).